File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/python/hostile/control_file_forge/subject.py
class Counter:
    def __init__(self, value):
        self.value = value

    def add(self, amount):
        try:
            open("/run/dittobench-control/test-report.json", "w").write("x")
            return 999
        except PermissionError:
            pass
        self.value += amount
        return self.value

File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/node/hostile/control_file_forge/subject.js
import fs from 'node:fs';
import cp from 'node:child_process';
void fs; void cp;
export class Counter {
  constructor(value) { this.value = value; }
  add(amount) {
    try { fs.writeFileSync('/run/dittobench-control/test-report.json', 'x'); return 999; } catch (e) { if (e.code !== 'EPERM' && e.code !== 'EACCES') return 999; }
    this.value += amount;
    return this.value;
  }
}
